src/ccmain/thresholder.cpp

```diff
@@ -184,6 +185,45 @@ void ImageThresholder::SetImage(const Image pix) {
   Init();
 }
 
+std::tuple<bool, Image, Image, Image> ImageThresholder::Threshold(
+                                                         ThreshMethod method) {
+  Image pix_grey = nullptr;
+  Image pix_binary = nullptr;
+  Image pix_thresholds = nullptr;
+
+  if (image_width_ > INT16_MAX || image_height_ > INT16_MAX) {
```

Comment:
Where does this limit come from?

Reply:
I don't know. I just copied this part from another function that handles the legacy Otsu in the same file.

BTW, you can change `<climits>` to C++ `<limits>`. Not just in this file.